test/e2e/query_test.go

```diff
@@ -631,6 +638,93 @@ func newSample(s fakeMetricSample) model.Sample {
 	}
 }
 
+// Regression test for https://github.com/thanos-io/thanos/issues/5033.
+// Tests whether queries work with mixed sources w/ max_over_time and
+// min_over_time.
+func TestSidecarStorePushdown(t *testing.T) {
+	t.Parallel()
+
+	// Build up.
+	e, err := e2e.NewDockerEnvironment("e2e_sidecar_store_pushdown")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, e))
+
+	prom1, sidecar1, err := e2ethanos.NewPrometheusWithSidecar(e, "p1", defaultPromConfig("p1", 0, "", ""), "", e2ethanos.DefaultPrometheusImage(), "remote-write-receiver")
```

Comment:
Is it intended to enable remote write receiver?

Reply:
The test synthesizes samples by remote writing them, which is why the receiver feature is needed.

Reply:
Thanks @fpetkovski. I thought it was using data from the block only. Didn't notice the remote write part.